libdeno/api.cc

```diff
@@ -225,4 +228,22 @@ void deno_terminate_execution(Deno* d_) {
   deno::DenoIsolate* d = reinterpret_cast<deno::DenoIsolate*>(d_);
   d->isolate_->TerminateExecution();
 }
+
+deno_buf deno_generate_snapshot(const char* js_file, const char* js_source) {
+  CHECK_NOT_NULL(js_file);
+  CHECK_NOT_NULL(js_source);
+  deno_init();
+  deno_config config = {1, deno::empty_buf, deno::empty_buf, nullptr};
+  Deno* d = deno_new(config);
+  deno_execute(d, nullptr, js_file, js_source);
+  if (deno_last_exception(d) != nullptr) {
+    std::cerr << "Snapshot Exception " << std::endl;
+    std::cerr << deno_last_exception(d) << std::endl;
+    deno_delete(d);
+    exit(1);  // TODO(afinch7): better error reporting here
+  }
+  auto snapshot = deno_get_snapshot(d);
+  deno_delete(d);
```

Comment:
This method is essentially a combination of `deno_init`, `deno_new`, `deno_execute`, `deno_last_exception`, `deno_delete`, and `deno_get_snapshot` - which prints the error to stderr and exits.

Since it does not introduce any new V8 bindings, I do not think this should be a top-level libdeno function.

Can you remove this from deno.h ?